### PR TITLE
[SPARK-12641] Remove unused code related to Hadoop 0.23

### PR DESCRIPTION
**Pull request by sarutak**
Currently we don't support Hadoop 0.23 but there is a few code related to it so let's clean it up.

### CODE REVIEW

File: core/src/main/scala/org/apache/spark/util/Utils.scala [hunk 1]
```diff
@@ -740,17 +738,12 @@ private[spark] object Utils extends Logging {
...
 
   /** Get the Yarn approved local directories. */
   private def getYarnLocalDirs(conf: SparkConf): String = {
-    // Hadoop 0.23 and 2.x have different Environment variable names for the
-    // local dirs, so lets check both. We assume one of the 2 is set.
-    // LOCAL_DIRS => 2.X, YARN_LOCAL_DIRS => 0.23.X
-    val localDirs = Option(conf.getenv("YARN_LOCAL_DIRS"))
-      .getOrElse(Option(conf.getenv("LOCAL_DIRS"))
-      .getOrElse(""))
+    val localDirs =Option(conf.getenv("LOCAL_DIRS")).getOrElse("")
```
Comment (rxin) [inline core/src/main/scala/org/apache/spark/util/Utils.scala]: add a space after = otherwise it will fail style test

File: scalastyle-config.xml [hunk 2]
```diff
@@ -218,6 +218,11 @@ This file is divided into 3 sections:
     </parameters>
   </check>
 
+  <check level="warning" class="org.scalastyle.scalariform.EnsureSingleSpaceBeforeTokenChecker" enabled="true">
+    <parameters>
+      <parameter name="tokens">LBRACE</parameter>
```
Comment (rxin) [inline scalastyle-config.xml]: this doesn't belong here?

Comment (sarutak) [inline scalastyle-config.xml]: Sorry, it's mistake. I'll remove it.

**Comment by rxin** [comment]
LGTM

**Comment by rxin** [comment]
I've merged this. Thanks.

**Pull request closed without merging**